chore: update docs for `dedupe` (#522)
tailcallhq/tailcallhq.github.io: docs/directives.md

This way you can compose combine multiple operations can compose them together using the `@call` directive.

### dedupe

> This flag when set to `true` prevents duplicate HTTP requests from being executed concurrently, reducing resource load. If not specified, this feature defaults to `false`.

Reviewer: ```suggestion
This flag when set to `true` prevents duplicate calls from being executed concurrently, reducing resource load. If not specified, this feature defaults to `false`.
```

> A boolean flag, if set to `true`, will enable deduplication of IO operations to enhance performance. This flag prevents duplicate IO requests from being executed concurrently, reducing resource load. If not specified, this feature defaults to `false`.

Reviewer: ```suggestion
This flag when set to `true` prevents duplicate calls from being executed concurrently, reducing resource load. If not specified, this feature defaults to `false`.
```

Make sure you have also specified batch settings to the `@upstream` and to the `@graphQL` directive.

### dedupe

> This flag when set to `true` prevents duplicate HTTP requests from being executed concurrently, reducing resource load. If not specified, this feature defaults to `false`.

Reviewer: ```suggestion
This flag when set to `true` prevents duplicate GraphQL requests from being executed concurrently, reducing resource load. If not specified, this feature defaults to `false`.
```

> A boolean flag, if set to `true`, will enable deduplication of IO operations to enhance performance. This flag prevents duplicate IO requests from being executed concurrently, reducing resource load. If not specified, this feature defaults to `false`.

Reviewer: ```suggestion
This flag when set to `true` prevents duplicate GraphQL requests from being executed concurrently, reducing resource load. If not specified, this feature defaults to `false`.
```

}
```

### dedupe

> This flag when set to `true` prevents duplicate HTTP requests from being executed concurrently, reducing resource load. If not specified, this feature defaults to `false`.

Reviewer: ```suggestion
This flag when set to `true` prevents duplicate gRPC requests from being executed concurrently, reducing resource load. If not specified, this feature defaults to `false`.
```

> A boolean flag, if set to `true`, will enable deduplication of IO operations to enhance performance. This flag prevents duplicate IO requests from being executed concurrently, reducing resource load. If not specified, this feature defaults to `false`.

Reviewer: ```suggestion
This flag when set to `true` prevents duplicate gRPC requests from being executed concurrently, reducing resource load. If not specified, this feature defaults to `false`.
```

}
```

### dedupe

> A boolean flag, if set to `true`, will enable deduplication of IO operations to enhance performance. This flag prevents duplicate IO requests from being executed concurrently, reducing resource load. If not specified, this feature defaults to `false`.

Reviewer: ```suggestion
This flag when set to `true` prevents duplicate HTTP requests from being executed concurrently, reducing resource load. If not specified, this feature defaults to `false`.
```

A boolean flag, if set to `true`, will enable deduplication of IO operations to enhance performance. This flag prevents duplicate IO requests from being executed concurrently, reducing resource load. If not specified, this feature defaults to `false`.

This feature is supported in `IO` based directives such as

> [`@http`](#http-directive), [`@grpc`](#grpc-directive), [`@graphQL`](#graphql-directive) and [`@call`](#call-directive)

Reviewer: This is defined under server. Lets delete it from here.

Reviewer: And add it in all the IO operations.
